### PR TITLE
Crew radar

### PR DESCRIPTION
**Pull request by 10Dozen**
- Adds extra settings for radar and indicators while in vehicle crew (gunner, driver, commander - so it should make vehicle formation easier to keep).
- Changes calculation of indicator icon size and warning zone to support long ranges.

### CODE REVIEW

**Comment by diwako** [comment]
I personally do like the idea of it, how ever I can see some units not wanting this. Would it be possible to add a way to force this feature off? Preferably some boolean.

Also would you like to add yourself to the `authors.txt`?

**Comment by 10Dozen** [comment]
> I personally do like the idea of it, how ever I can see some units not wanting this. Would it be possible to add a way to force this feature off? Preferably some boolean.

TBH i was thinking that they can just force crew range same as non-crew (or set default values to be the same for both ranges). But extra bool setting is fine too.

**Comment by 10Dozen** [comment]
Added checkbox options for both modules.

File: addons/main/functions/fnc_isInCrew.sqf [hunk 1]
```diff
@@ -0,0 +1,7 @@
+#include "script_component.hpp"
+if (is3DEN || !hasInterface) exitWith { false };
+
+params [["_player", objNull]];
+private _vehicle = vehicle _player;
+
+(_vehicle != _player) && { fullCrew [_vehicle, "cargo"] findIf {_player == _x # 0} == -1 }
```
Comment (diwako) [inline addons/main/functions/fnc_isInCrew.sqf:7]: Please change the `#` to `select`. The linter hates this as `#` is a pre processor thing.
```suggestion
(_vehicle != _player) && { fullCrew [_vehicle, "cargo"] findIf {_player isEqualTo (_x select 0)} isEqualTo -1 }
```

File: addons/radar/XEH_preInit.sqf [hunk 2]
```diff
@@ -197,6 +215,15 @@ private _curCat = localize "STR_dui_cat_compass";
     ,false
 ] call CBA_fnc_addSetting;
 
+[
+    "diwako_dui_compass_icon_scale_crew"
```
Comment (diwako) [inline addons/radar/XEH_preInit.sqf:219]: Please use the QGVAR macro

File: addons/radar/XEH_preInit.sqf [hunk 3]
```diff
@@ -141,6 +141,24 @@ private _curCat = localize "STR_dui_cat_compass";
...
+    QGVAR(vehicleCompassEnabled)
+    ,"CHECKBOX"
+    ,[localize "STR_dui_compass_vehicle_enabled", localize "STR_dui_compass_vehicle_enabled_desc"]
+    ,[CBA_SETTINGS_CAT, _curCat]
+    ,false
+    ,true
+] call CBA_fnc_addSetting;
+
+[
+    "diwako_dui_compassRangeCrew"
```
Comment (diwako) [inline addons/radar/XEH_preInit.sqf:154]: Please use the QGVAR macro

File: addons/radar/XEH_preInit.sqf (hunk 2 above)
Comment (10Dozen) [reply addons/radar/XEH_preInit.sqf:219]: Only these or all option varnames in the file?

Comment (diwako) [reply addons/radar/XEH_preInit.sqf:219]: For the newly added ones. The reason why not all variable names use the GVAR macros are for backwards compatibility before this was a cba template mod.
All new settings should use the macros now.

Comment (10Dozen) [reply addons/radar/XEH_preInit.sqf]: Replaced, now new global vars are using macro

File: addons/radar/functions/fnc_displayUnitOnCompass.sqf [hunk 4]
```diff
@@ -28,6 +28,15 @@ private _iconScale = diwako_dui_compass_icon_scale;
 private _distanceWarning = diwako_dui_distanceWarning;
 private _showSpeaking = GVAR(showSpeaking);
 
+if (GVAR(vehicleCompassEnabled) && { _player call EFUNC(main,isInCrew) }) then {
+    _circleRange = GVAR(compassRangeCrew);
+    _iconScale = GVAR(icon_scale_crew);
+    _distanceWarning = linearConversion [
+        DUI_MIN_VEHICLE_RANGE, DUI_MAX_VEHICLE_RANGE, diwako_dui_compassRangeCrew,
```
Comment (diwako) [inline addons/radar/functions/fnc_displayUnitOnCompass.sqf:35]:
```suggestion
        DUI_MIN_VEHICLE_RANGE, DUI_MAX_VEHICLE_RANGE, GVAR(compassRangeCrew),
```
`diwako_dui_compassRangeCrew` was replaced with a macro

File: addons/radar/XEH_preInit.sqf [hunk 5]
```diff
@@ -197,6 +215,15 @@ private _curCat = localize "STR_dui_cat_compass";
     ,false
 ] call CBA_fnc_addSetting;
 
+[
+    QGVAR(icon_scale_crew)
+    ,"SLIDER"
+    ,[localize "STR_dui_compass_icon_scale_crew", localize "STR_dui_compass_icon_scale_crew_desc"]
+    ,[CBA_SETTINGS_CAT, _curCat]
+    ,[0.01, 24, 6, 2]
```
Comment (diwako) [inline addons/radar/XEH_preInit.sqf:223]: After checking it out, 6 is quite large, at least on 1080p with small UI scaling.

Comment (diwako) [reply addons/radar/XEH_preInit.sqf:223]: OH, I see. It does scale with the crew range. It is find on 6 and 500 meters range, my apologies.

**Pull request merged**